src/faction.cpp

```diff
@@ -1053,3 +1088,308 @@ void faction_manager::display() const
+                        if( guy->current_target() != nullptr ) {
+                            current_status += _( "In Combat!" );
+                            status_col = c_light_red;
+                        } else if( guy->in_sleep_state() ) {
+                            current_status += _( "Sleeping" );
+                        } else if( guy->is_following() ) {
+                            current_status += _( "Following" );
+                        } else if( guy->is_leader() ) {
+                            current_status += _( "Leading" );
+                        } else if( guy->is_guarding() ) {
+                            current_status += _( "Guarding" );
+                        }
+                        mvwprintz( w_missions, ++y, 31, status_col, current_status );
+                        const std::pair <std::string, nc_color> condition = guy->hp_description();
+                        const std::string condition_string = _( "Condition : " ) + condition.first;
+                        mvwprintz( w_missions, ++y, 31, condition.second, condition_string );
+                        const std::pair <std::string, nc_color> hunger_pair = guy->get_hunger_description(
+                                    guy->get_hunger() );
+                        const std::pair <std::string, nc_color> thirst_pair = guy->get_thirst_description(
+                                    guy->get_thirst() );
+                        const std::pair <std::string, nc_color> fatigue_pair = guy->get_fatigue_description(
+                                    guy->get_fatigue() );
+                        mvwprintz( w_missions, ++y, 31, hunger_pair.second,
+                                   _( "Hunger : " ) + ( ( hunger_pair.first == "" ) ? "Nominal" : hunger_pair.first ) );
+                        mvwprintz( w_missions, ++y, 31, thirst_pair.second,
+                                   _( "Thirst : " ) + ( ( thirst_pair.first == "" ) ? "Nominal" : thirst_pair.first ) );
+                        mvwprintz( w_missions, ++y, 31, fatigue_pair.second,
+                                   _( "Fatigue : " ) + ( ( fatigue_pair.first == "" ) ? "Nominal" : fatigue_pair.first ) );
+                        const std::string wield_str = _( "Wielding : " ) + guy->weapon.tname();
+                        int lines = fold_and_print( w_missions, ++y, 31, getmaxx( w_missions ) - 33, c_white,
+                                                    ( wield_str ) );
+                        y += lines;
+                        const auto skillslist = Skill::get_skills_sorted_by( [&]( const Skill & a, const Skill & b ) {
+                            const int level_a = guy->get_skill_level( a.ident() );
+                            const int level_b = guy->get_skill_level( b.ident() );
+                            return level_a > level_b || ( level_a == level_b && a.name() < b.name() );
+                        } );
+                        int index_one = -1;
```

Comment:
strongly consider doing:
```
size_t count = 0;
std::vector<std::string> skill_strs;
for( int i = 0; i < static_cast<int>( skillslist.size() ) && count < 3; i++ ) {
    if( !skillslist[ i ]->is_combat_skill() ) {
        std::string skill_str = skillslist[i]->name() + " : " + std::to_string( guy->get_skill_level( skillslist[i]->indent() );
        skills_strs.push_back( skill_str );
        count += 1;
    }
}
mvwprintz( w_missions, ++y, 31, col, best_three_noncombat + skill_strs[0] );
mvwprintz( w_missions, ++y, 51, col, skill_strs[1] );
mvwprintz( w_missions, ++y, 51, col, skill_strs[2] );
```
it's a lot shorter and easier to read, plus there's no need to loop through stuff repeatedly.